Rubberduck.Inspections/VariableRequiresSetAssignmentEvaluator.cs

```diff
@@ -21,37 +22,115 @@ public static IEnumerable<Declaration> GetDeclarationsPotentiallyRequiringSetAss
 
         public static bool RequiresSetAssignment(IdentifierReference reference, RubberduckParserState state)
         {
-            //Not an assignment...definitely does not require a 'Set' assignment
             if (!reference.IsAssignment)
             {
+                // reference isn't assigning its declaration; not interesting
                 return false;
             }
-            
-            //We know for sure it DOES NOT use 'Set'
-            if (!MayRequireAssignmentUsingSet(reference.Declaration))
+
+            var setStmtContext = reference.Context.GetAncestor<VBAParser.SetStmtContext>();
+            if (setStmtContext != null)
+            {
+                // assignment already has a Set keyword
+                return true;
+                // (but is it misplaced? ...hmmm... beyond the scope of *this* inspection though)
+                // if we're only ever assigning to 'Nothing', might as well flag it though
+                if (reference.Declaration.References.Where(r => r.IsAssignment).All(r =>
+                    r.Context.GetAncestor<VBAParser.SetStmtContext>().expression().GetText() == Tokens.Nothing))
+                {
+                    return true;
+                }
+            }
+
+            var letStmtContext = reference.Context.GetAncestor<VBAParser.LetStmtContext>();
+            if (letStmtContext == null)
+            {
+                // we're probably in a For Each loop
+                return false;
```

Comment:
I think the check should be tightened to explicitly check that we are actually inside a `For Each` to avoid generating wrong results.